extensions/schema-compare/src/schemaCompareMainWindow.ts

```diff
@@ -159,14 +163,38 @@ export class SchemaCompareMainWindow {
 			};
 		}
 
+		await this.launch(source, target, false, comparisonResult);
+	}
+
+	/**
+	 * Primary functional entrypoint for opening the schema comparison window, and optionally running it.
+	 * @param source
+	 * @param target
+	 * @param runComparison whether to immediately run the schema comparison.  Requires both source and target to be specified.  Cannot be true when comparisonResult is set.
+	 * @param comparisonResult a pre-computed schema comparison result to display.  Cannot be set when runComparison is true.
+	 */
+	public async launch(source: mssql.SchemaCompareEndpointInfo | undefined, target: mssql.SchemaCompareEndpointInfo | undefined, runComparison: boolean = false, comparisonResult: mssql.SchemaCompareResult | undefined) {
+		if (runComparison && comparisonResult) {
+			throw new Error('Cannot both pass a comparison result and request a new comparison be run.');
```

Comment:
Do these error messages not need to be localized because they won't be shown to the user?

Reply:
Correct.